README.md

```diff
@@ -1,58 +1,195 @@
+
+### Using a custom NPM registry
+
+Any of the keys in [options.yml](blob/master/config/options.yml) can be overridden.
 
-   - Create a local directory with an `options.yml` with the configuration changes
-     you want to make, then when running the container image, mount that directory
-     with `-v <dir>:/root/OPTIONS`.
    - If overriding the NPM registry, set the `npm_registry` key in the `options.yml`.
    - If building RPMs in Copr,
      - set the `rpm.repo_name` key in the `options.yml`.
      - run the container image with `-v <copr token file>:/root/.config/copr`.
    - If updating the RPM repo it would be helpful to attach a volume to hold the RPM cache with `-v <dir>:/root/rpm_cache`.
      Any RPMs not in the cache will be downloaded to the cache first.
 
-3. Start the container image
+### Artifacts
+
+The purpose of this process is to build code from `manageiq-core`, `manageiq-gemset`, and `manageiq-appliance` into 2 sets of artifacts:
 
-   `docker run -it <image>`
+- `/root/BUILD/rpm_spec/*.tar.gz`
+- `/root/BUILD/rpms/x86_64/manageis-*.rpm`
 
-4. Run the script
+Depending upon your usecase, there are a few ways to obtain these artifacts:
 
-   - Run `bin/build.rb`, optionally overriding git ref (branch or tag) and release type
+- Have the container put the artifacts into a mounted volume.
+- Manually copy the artifacts to a mounted volume.
+- Extracting the artifacts from the docker container with a docker command.
 
-     ```
-     bin/build.rb --git-ref jansa --update-rpm-repo
-     ```
-     ```
-     bin/build.rb --build-type release --git-ref jansa-1-alpha1 --update-rpm-repo
-     ```
+### Having the container build files into a mounted volume
 
-Alternatively, the build can be started directly by passing `build` to the
-`docker run` command:
+Since the files are not uploaded, they need to be retrieved out of the repo before the container exits.
 
+Note: all mounted volumes (i.e.: using the `-v` option) need to be defined in docker's preferences:
+
+```
+Settings > Resources > FILE SHARING
+```
+
+For this example, 2 directories exist in the current directory (`pwd`):
+
+- `OPTIONS` a directory contains the `options.yml` file
+- `BUILD` a directory that will contain the `tar.gz` and `rpm` files.
+
+Note: The contents of `BUILD` will be deleted before each run.
+
+```sh
+docker run -v --rm -v `pwd`/OPTIONS:/root/OPTIONS -v `pwd`/BUILD:/root/BUILD manageiq/rpm_build:latest build
+```
+
+### Manually copying files to a mounted volume
+
+Since defining the mount points can be inconvenient, it is possible to just run a build and copy out the files out.
+The trick is to run the container as a daemon and in a way that it does not exit before you have copied out the resources.
+
+#### Running the container in daemon mode
+
+```sh
+# setup docker and remember the CONTAINER id
+CONTAINER=`docker run -d -it -v $(pwd)/OPTIONS:/root/OPTIONS manageiq/rpm_build:latest`
```

Comment:
it's generally a lot easier if you're running a daemonized container to give it a name using `--name`.  Then later you just reference the container by name directly, and then no need for env vars or anything.